feat(commands/mr/create): use tracking branch as target branch if not given

maxice8 opened the pull request:
**Description**
<!--- Describe your changes in detail -->
If --target-branch is not given and there is tracking branch, use it.

**Related Issue**
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
Resolves #582 

**How Has This Been Tested?**
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Tested against `gitlab.alpinelinux.org`

**Types of changes**
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation
- [ ] Chore (Related to CI or Packaging to platforms)

maxice8 committed:
feat(commands/mr/create): use tracking branch as target branch if not given

profclems commented on commands/mr/create/mr_create.go:
```diff
@@ -209,6 +204,20 @@ func NewCmdCreate(f *cmdutils.Factory) *cobra.Command {
 
+			if opts.TargetBranch == "" {
+				// Check if have a tracking branch (MergeRef) and use it as target branch
+				branchConfig := git.ReadBranchConfig(opts.SourceBranch)
+				if branchConfig.RemoteName != "" && branchConfig.MergeRef != "" {
```
`git.ReadBranchConfig` is likely to return a `nil` if an error occurs. See

https://github.com/profclems/glab/blob/b2922a72dcf07f25a95046cdd3e9ff0afc2e1ff3/internal/git/git.go#L234-L240

 so it is important to check if `branchConfig` is not nil before accessing the fields to avoid panic

maxice8 replied:
I tested if it returns something that isn't `(git.BranchConfig{})`, hope that is enough

profclems reviewed (changes requested):
Thanks for working on this.

I have a little issue here:

profclems commented on commands/mr/create/mr_create.go:
```diff
@@ -209,6 +204,24 @@ func NewCmdCreate(f *cmdutils.Factory) *cobra.Command {
+				// Check if our given git.BranchConfig{} is not empty, otherwise it will fail
+				// if try to access the fields, this is needed because the ReadBranchConfig
+				// function can return an empty struct
+				if branchConfig != (git.BranchConfig{}) {
+					if branchConfig.RemoteName != "" && branchConfig.MergeRef != "" {
+						// The MergeRef takes the form of refs/head/BRANCH_NAME, so split it
+						// by '/' and get the last element
+						branchName := strings.Split(branchConfig.MergeRef, "/")
+						opts.TargetBranch = branchName[len(branchName)-1]
+					}
+				} else {
+					opts.TargetBranch, _ = git.GetDefaultBranch(baseRepoRemote.PushURL.String())
+				}
```
```suggestion
				if branchConfig != nil {
					if branchConfig.RemoteName != "" && branchConfig.MergeRef != "" {
						// The MergeRef takes the form of refs/head/BRANCH_NAME, so split it
						// by '/' and get the last element
						branchName := strings.Split(branchConfig.MergeRef, "/")
						opts.TargetBranch = branchName[len(branchName)-1]
					} else {
						opts.TargetBranch, _ = git.GetDefaultBranch(baseRepoRemote.PushURL.String())
					}
				} else {
					opts.TargetBranch, _ = git.GetDefaultBranch(baseRepoRemote.PushURL.String())
				}
```

profclems committed:
add function to get target branch

Refactored the part of the code in runE that determines the target branch
and created a separate function for it.
This will help test the function separately

profclems committed:
test(commands/mr/create): print actual error from stderr if error occurs

profclems committed:
fix tests
